packages/untp-playground/src/lib/schemaValidation.ts

```diff
@@ -18,21 +19,117 @@ const SCHEMA_URLS = {
   DigitalIdentityAnchor: 'https://test.uncefact.org/vocabulary/untp/dia/untp-dia-schema-0.2.1.json',
 };
 
+const EXTENSION_VERSIONS: Record<
+  string,
+  { domain: string; versions: { version: string; core: { type: string; version: string } }[] }
+> = {
+  DigitalLivestockPassport: {
+    domain: 'aatp.foodagility.com',
+    versions: [
+      {
+        version: '0.4.0',
+        core: { type: 'DigitalProductPassport', version: '0.5.0' },
+      },
+    ],
+  },
+};
+
+const schemaURLConstructor = (type: string, version: string) => {
+  const shortCredentialTypes: Record<string, string> = {
+    DigitalProductPassport: 'dpp',
+    DigitalConformityCredential: 'dcc',
+    DigitalTraceabilityEvent: 'dte',
+    DigitalFacilityRecord: 'dfr',
+    DigitalIdentityAnchor: 'dia',
+  };
+  return `https://test.uncefact.org/vocabulary/untp/${shortCredentialTypes[type]}/untp-${shortCredentialTypes[type]}-schema-${version}.json`;
+};
+
+const extensionSchemaURLConstructor = (type: string, version: string) => {
+  const shortCredentialTypes: Record<string, string> = {
+    DigitalLivestockPassport: 'dlp',
+  };
+  if (type === 'DigitalLivestockPassport' && version === '0.4.0') {
+    return 'https://aatp.foodagility.com/assets/files/aatp-dlp-schema-0.4.0-9c0ad2b1ca6a9e497dedcfd8b87f35f1.json';
+  }
+  return `https://aatp.foodagility.com/vocabulary/aatp/${shortCredentialTypes[type]}/aatp-${shortCredentialTypes[type]}-schema-${version}.json`;
+};
```

Comment:
I think we can create JSON files to map credential types to their schema URLs using the format { credentialType: credentialContextURL }. For example:

UNTP schema URLs:

```json
{
    "DigitalFacilityRecord": "https://test.uncefact.org/vocabulary/untp/dfr/0.5.0/",
    "DigitalIdentityAnchor": "https://test.uncefact.org/vocabulary/untp/dia/0.2.1/",
    //...Others
}
```


Extension schema URLs:

```json
{
    "DigitalLivestockPassport": "https://aatp.foodagility.com/assets/files/aatp-dlp-schema-0.4.0-9c0ad2b1ca6a9e497dedcfd8b87f35f1.json",
    //...Others
}
```

If we go this way, we can avoid writing custom logic to construct the schema URLs, including extension schema URLs. Since in the future, we're not sure they are the same structure. Additionally, if schema URLs change in the future, we can simply update the json mapping file instead of modifying the codebase, making the maintenance more straightforward.

Reply:
I will refactor it to merge the configuration data together.